stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModelBinding.kt

```diff
@@ -37,25 +39,23 @@ public fun ChannelListViewModel.bindView(
     view: ChannelListView,
     lifecycleOwner: LifecycleOwner,
 ) {
-    state.observe(lifecycleOwner) { channelState ->
-        if (channelState.isLoading) {
-            view.showLoadingView()
-        } else {
-            view.hideLoadingView()
-            channelState
-                .channels
-                .map(ChannelListItem::ChannelItem)
-                .let(view::setChannels)
+    state.combineWith(paginationState) { state, paginationState ->
+        paginationState?.let {
+            view.setPaginationEnabled(!it.endOfChannels && !it.loadingMore)
         }
-    }
 
-    paginationState.observe(lifecycleOwner) { paginationState ->
-        view.setPaginationEnabled(!paginationState.endOfChannels && !paginationState.loadingMore)
+        var list: List<ChannelListItem> = state?.channels?.map(ChannelListItem::ChannelItem) ?: emptyList()
+        if (paginationState?.loadingMore == true) {
+            list = list + ChannelListItem.LoadingMoreItem
+        }
```

Comment:
I logged whats going on and seems that loading a new page produced redundant emit. Given 30 loaded channels and the list is scrolled to the end:

1. 30 channels + 1 loading more item
2. 60 channels + 1 loading more item <- autoscrolling here to the 60th channel
3. 60 channels 

I think the list should be scrolled to the 30th item (or 31 item in case the progress bar is seamlessly replaced with the 31 item). Could you check?

Reply:
Yes, seems that the loadingMore flag was not set to false before so the indicator would be added and scrolled to. Can you chek out the fix?